### ISSUE TITLE
Fetch listing via IPFS and sanitize

### ISSUE THREAD

**Issue by drwasho** [OpenBazaar/openbazaar-go#1444]
There are a few ways to retrieve a listing from the network:

1. `GET /ipfs/:listingHash`
2. `GET /ipns/:peerId/:slug_or_listingHash`
3. `GET /ob/listing/:listingHash`
4. `GET /ob/listing/:peerId/:slug_or_listingHash`

For API calls 1-2, these retrieve the JSON files _as is_. In contrast, API calls 3-4 retrieve the listing and sanitize the data (adding some important information).

For API call (3), the server looks at the node's own filesystem to retrieve a listing the corresponds to the listing hash or slug. Not including a `peerId` in the URL path of this call is how we tell the server to look for one of its own listings, and not an external node's listing.

To fetch external listings, you need to use API call (4), which uses IPNS to retrieve the data.

**tl;dr** - there is no way to retrieve a listing using IPFS where the server sanitizes it in the same way it does for listings retrieved locally or via IPNS in API calls (3) and (4) respectively. If we want to significantly enhance the speed that we can retrieve listings via IPFS, we need to be able to sanitize these listings.

**Comment by cpacia** [comment]
4) does an ipfs call if you pass in a hash, not an ipns call. it only does an ipns call if you use the slug

**Comment by drwasho** [comment]
FYI to anyone coming across this issue, you need to make this call:

`GET /ob/listing/ipfs/:hash`
